src/Collection.php

```diff
@@ -244,34 +250,84 @@ public function count(array $filter = array(), array $options = array())
     }
 
     /**
-     * Create a single index in the collection.
+     * Create a single index for the collection.
      *
      * @see http://docs.mongodb.org/manual/reference/command/createIndexes/
      * @see http://docs.mongodb.org/manual/reference/method/db.collection.createIndex/
-     * @param array|object $keys
-     * @param array        $options
+     * @see Collection::createIndexes()
+     * @param array|object $key     Document containing fields mapped to values,
+     *                              which denote order or an index type
+     * @param array        $options Index options
      * @return string The name of the created index
      */
-    public function createIndex($keys, array $options = array())
+    public function createIndex($key, array $options = array())
     {
-        // TODO
+        return current($this->createIndexes(array(array('key' => $key) + $options)));
     }
 
     /**
-     * Create multiple indexes in the collection.
+     * Create one or more indexes for the collection.
+     *
+     * Each element in the $indexes array must have a "key" document, which
+     * contains fields mapped to an order or type. Other options may follow.
+     * For example:
      *
-     * TODO: decide if $models should be an array of associative arrays, using
-     * createIndex()'s parameter names as keys, or tuples, using parameters in
-     * order (e.g. [keys, options]).
+     *     $indexes = [
+     *         // Create a unique index on the "username" field
+     *         [ 'key' => [ 'username' => 1 ], 'unique' => true ],
+     *         // Create a 2dsphere index on the "loc" field with a custom name
+     *         [ 'key' => [ 'loc' => '2dsphere' ], 'name' => 'geo' ],
+     *     ];
+     *
+     * If the "name" option is unspecified, a name will be generated from the
+     * "key" document.
      *
      * @see http://docs.mongodb.org/manual/reference/command/createIndexes/
      * @see http://docs.mongodb.org/manual/reference/method/db.collection.createIndex/
-     * @param array $models
+     * @param array $indexes List of index specifications
      * @return string[] The names of the created indexes
+     * @throws InvalidArgumentException if an index specification does not
+     *                                  contain a "key" document
+     * @throws UnexpectedTypeException if an index specification is not an array
+     *                                 or a "key" document is not an array or
+     *                                 object
      */
-    public function createIndexes(array $models)
+    public function createIndexes(array $indexes)
     {
-        // TODO
+        foreach ($indexes as &$index) {
```

Comment:
If `$indexes` is empty, we should raise an InvalidArgumentException because this is a NOP otherwise.